src/lib/linked-list.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const Node = require('./node');
+
+module.exports = class LinkedList {
+  constructor() {
+    this.length = 0;
+    this.head = null;
+  }
+
+  // Big O = O(1)
+  insertAtHead(value) {
+    const node = new Node(value);
+
+    node.next = this.head;
+    this.head = node;
+    this.length += 1;
+    return this;
+  }
+
+  // Big O = O(n)
+  find(value) {
+    if (!this.head) {
+      throw new Error('__ERROR__ The list is empty');
+    }
+
+    if (this.head.value === value) {
+      return this.head;
+    }
+
+    let currentNode = this.head;
+    while (currentNode.next) {
+      if (currentNode.next.value === value) {
+        return currentNode.next;
+      }
+      currentNode = currentNode.next;
+    }
+    return null;
+  }
+
+  // Big O = O(n)
+  pop(position) {
+    if (!this.head) {
+      throw new Error('__ERROR__ Empty list');
+    } 
+    if (this.head.value === position) {
+      this.head = this.head.next;
+    } else {
+      let previousNode = this.head;
+      let currentNode = previousNode.next;
+      while (currentNode) {
+        if (currentNode.value === position) {
+          previousNode.next = currentNode.next;
+          currentNode = currentNode.next;
+          break;
+        } else {
+          previousNode = currentNode;
+          currentNode = currentNode.next;
+        }
+      }
+    }
+    return this;
+  }
+
+  // Big o = O(n)
+  reduce() {
```

Comment:
I won't take off points for this, but remember that `reduce` is used for more than just summing values, it REDUCES a collection to a whole new value, which can be anything, such as a sum of values, an entirely new object, a whole new array, or a whole new linked list. 

Remember it takes a callback in order to do that. See the MDN docs for the Arrray's reduce method. I recommend taking a look at the polyfill section on the MDN page to understand how reduce works under the hood.

You don't need to make your reduce THAT complex, but hopefully taking a look at it can push you the right way toward writing a more robust reduce for Linked Lists.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce